Modules/_testcapimodule.c

```diff
@@ -2385,7 +2385,11 @@ test_null_strings(PyObject *self, PyObject *Py_UNUSED(ignored))
     PyObject *tuple = PyTuple_Pack(2, o1, o2);
     Py_XDECREF(o1);
     Py_XDECREF(o2);
-    return tuple;
+    Py_XDECREF(tuple);
+    if (tuple == NULL) {
```

Comment:
BTW, comparing a pointer to freed memory with NULL is undefined behavior. `Py_DECREF(tuple)` should be after `if (tuple == NULL)`.

Reply:
Looks like I've forgotten what I knew of C; thanks for reviewing and catching this!